core/cloudflow-operator-actions/src/main/scala/cloudflow/operator/action/EndpointActions.scala

```diff
@@ -50,12 +50,17 @@ object EndpointActions {
 
     val deleteActions = (currentEndpoints -- newEndpoints).flatMap { endpoint ⇒
       Seq(
-        Action.delete[Service](Name.ofService(StreamletDeployment.name(newApp.spec.appId, endpoint.streamlet)), namespace)
+        Action.delete[Service](Name.ofService(StreamletDeployment.name(newApp.spec.appId, endpoint.streamlet)), newApp, namespace)
       )
     }.toList
     val createActions = (newEndpoints -- currentEndpoints).flatMap { endpoint ⇒
       Seq(
-        createServiceAction(endpoint, StreamletDeployment.name(newApp.spec.appId, endpoint.streamlet), namespace, labels, ownerReferences)
+        createServiceAction(endpoint,
+                            newApp,
```

Comment:
as well if we pass around the entire Cr this can be simpler maybe

Reply:
In this case we can cleanup a little bit, yes.